krm-functions/lib/kptrl/kptrl.go

```diff
@@ -17,65 +17,33 @@
 package kptrl
 
 import (
-	"sync"
-
 	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
 )
 
-type ResourceList interface {
-	// AddResult adds a result with error and corresponding kubeObject by
-	// appending a result to the result slice in the resourceList
-	AddResult(err error, obj *fn.KubeObject)
-	// GetResults gets the results slice from the resourceList
-	GetResults() fn.Results
-	// GetObject return an fn sdk KubeObject by comparing the APIVersion, Kind, Name and Namespace
-	// if the object is found the corresponding obj is returned, if not nil is returned
-	GetObject(obj *fn.KubeObject) *fn.KubeObject
-	// GetObjects returns all items from the resourceList
-	GetObjects() fn.KubeObjects
-	// SetObject sets the object in the resourceList items. It either updates/overrides
-	// the entry if it exists or appends the entry if it does not exist in the resourceList
-	// It uses APIVersion, Kind, Name and Namespace to check the object uniqueness
-	SetObject(obj *fn.KubeObject) error
-	// DeleteObject deletes the object from the resourceList if it exists.
-	DeleteObject(obj *fn.KubeObject)
+type ResourceList struct {
+	rl *fn.ResourceList
 }
 
-// New creates a new ResourceList interface
-// concurrency is handles in the methods
-func New(rl *fn.ResourceList) ResourceList {
-	return &resourceList{
+func New(rl *fn.ResourceList) *ResourceList {
+	return &ResourceList{
 		rl: rl,
 	}
 }
 
-// Typically in functions there is no concurrency, but this library may be used in specializers
-// running concurrently in the same binary, and may share the same resourceList".
-type resourceList struct {
-	m  sync.RWMutex
-	rl *fn.ResourceList
-}
-
 // AddResult adds a result with error and corresponding KubeObject by
 // appending a result to the result slice in the resourceList
-func (r *resourceList) AddResult(err error, obj *fn.KubeObject) {
-	r.m.Lock()
-	defer r.m.Unlock()
+func (r *ResourceList) AddResult(err error, obj *fn.KubeObject) {
 	r.rl.Results = append(r.rl.Results, fn.ErrorConfigObjectResult(err, obj))
 }
 
 // GetResults gets the results slice from the resourceList
-func (r *resourceList) GetResults() fn.Results {
-	r.m.RLock()
-	defer r.m.RUnlock()
+func (r *ResourceList) GetResults() fn.Results {
 	return r.rl.Results
 }
 
 // GetObject return an fn sdk KubeObject by comparing the APIVersion, Kind, Name and Namespace
 // if the object is found the corresponding obj is returned, if not nil is returned
-func (r *resourceList) GetObject(obj *fn.KubeObject) *fn.KubeObject {
-	r.m.RLock()
-	defer r.m.RUnlock()
+func (r *ResourceList) GetObject(obj *fn.KubeObject) *fn.KubeObject {
 	for _, o := range r.rl.Items {
```

Comment:
Just noticed this: https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn#KubeObjects.Where

This, along with change described above, would then become:

```

func (r *ResourceList) GetObject(obj *fn.KubeObject) *fn.KubeObject {
        return r.Items.Where(func(ko *KubeObject) bool { isGVKNNEqual(ko, obj) })
}

```

Reply:
where returns multiple objects AFAIS. So wanted to return 1 but will return all that match and use this

Reply:
Ah, right